### PR TITLE
[AMORO-2208] Validate NULL data expiration configuration

### LINKED ISSUE #2208: [Bug]: Throws NPE if the data expiration config is not serialized in the table configuration
### What happened?

AMS will keep throwing NPE if the data expiration is not serialized in table configuration in database, this will be occured when upgrading from versions lower than 0.6.0

### Affects Versions

master/0.6.0

### What engines are you seeing the problem on?

Core

### How to reproduce

1. keep old records in database
2. upgrade 0.6.0 from 0.5.x
3. enable data expiration

### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2208.

## Brief change log

- Avoid NPE

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by wangtaohz — APPROVED** [review]
LGTM

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java [hunk 1]
```diff
@@ -93,6 +93,9 @@ protected long getNextExecutingTime(TableRuntime tableRuntime) {
 
   @Override
   protected boolean enabled(TableRuntime tableRuntime) {
+    if (null == tableRuntime.getTableConfiguration().getExpiringDataConfig()) {
```
[the remark this answers is not in the document]
Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java:96]: The `TableConfiguration` ignores the fields that is not exist(via Jackson annotate), so expiration configuration will not be deserilized.

[the remark this answers is not in the document]
Comment (shidayang) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java:96]: Maybe getExpiringDataConfig return a default config if expiringDataConfig is null?

[the remark this answers is not in the document]
Comment (shidayang) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java:96]: Leaving the handling of null values to the user of TableConfiguration  may cause some difficulties in their usage.

[the remark this answers is not in the document]
Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java:96]: make sense

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
LGTM

**Pull request merged**